iroha/src/sumeragi/mod.rs

```diff
@@ -94,7 +99,8 @@ where
     events_sender: EventsSender,
     wsv: Arc<WorldStateView<W>>,
     /// Hashes of the transactions that were forwarded to a leader, but not yet confirmed with a receipt.
-    transactions_awaiting_receipts: Arc<DashSet<Hash>>,
+    /// And time at which this transaction was sent to the leader by this peer.
+    transactions_awaiting_receipts: Arc<DashMap<Hash, Duration>>,
```

Comment:
Kinda dont like wrapping fields in arc in times when we know that they will be alive.

Found this crate, which has api like crossbeam does, but for async (though unsafe):
https://docs.rs/async-scoped/0.7.0/async_scoped/index.html

May be we can do smth about that, as we know more?
Tokio seems to give up on building in this kind of scopes:
https://github.com/tokio-rs/tokio/issues/2596#issuecomment-663349217

Reply:
Could you create an issue for that?

Reply:
Or i guess we shouldn't, as iroha_p2p incoming